quadra/tasks/classification.py

```diff
@@ -423,6 +440,24 @@ def freeze_layers(self, freeze_parameters_name: List[str]):
 
         log.info("Frozen %d parameters", count_frozen)
 
+    def freeze_layers_by_index(self, freeze_parameters_index: List[int]):
+        """Freeze layers specified in freeze_parameters_name.
+
+        Args:
+            freeze_parameters_index: Layers that will be frozen during training.
+
+        """
+        count_frozen = 0
+        for i, (name, param) in enumerate(self.model.named_parameters()):
```

Comment:
Isn't it a bit risky to count on `named_parameters` since they don't exactly refer to `layers`?

Reply:
Any idea around that? I would avoid using named modules (also I think they suffer from the same problem), we could count only if they have weight in the name (can we have parameters with just bias?)

Alternative we can rename the function to freeze_parameters instead of freeze_layers so it would be more aligned to the user expectation

Reply:
I vote for renaming it to `freeze_parameters_by_index` at least it won't be misguiding. I don't know if there is a more elegant way to do it. But it will be always risky if a torch module (layer) has more than one parameter.